### PR TITLE
#162949222 Favourite and unfavourite articles

### PR DESCRIPTION
**Pull request by kelvinwm**
**What does this PR do?**
- Favouriting and unfavouriting articles

**Description of the Task to be completed**
- Favourite an article
- Unfavourite an article

**How should this be manually tested?**
- Clone the repository: git clone https://github.com/andela/ah-the-unsullied.git
- cd ah-the-unsullied
- git checkout ft-favourite-article-162949222
- create .env on the ah-the-unsullied folder and add the following
```export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/NAME'
     export SECRET_KEY="<your_secret_key_here>"
     export HOST_PASSWORD="<Gmail_Password>"
     export HOST_USER="<Gmail_Username>"
     export SMTP_HOST='smtp.gmail.com'
     export EMAIL_PORT='587
```
- Run the following commands 
```python3 -m venv venv
     source .env
     python manage.py migrate
     python manage.py runserver
```
- open postman and test the app
   
**Relevant PT story**
[#162949222](https://www.pivotaltracker.com/story/show/162949222)

**screenshots**
- N/A

### CODE REVIEW

**Review by kwanj-k — COMMENTED** [review]
Hey, good work so far, refactor the response to have keys may be 
e.g "message": "article successfully favourited!"
Also, the response after I get my favourites should be a little more detailed
like the title and slug/link to the article

**Review by kwanj-k — DISMISSED** [review]

**Review by nesh-dev — DISMISSED** [review]

File: authors/tests/test_favorite_article.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                                              body='iko sawa')
+        return res
+
+    def test_make_article_favorite(self):
+        """Test if user can favorite an article"""
+        res = self.add_article()
+        response = self.client.put(reverse('articles:favorite_article',
+                                           kwargs={'slug': 'kenya-moja'}),
+                                   content_type='application/json',
+                                   HTTP_AUTHORIZATION='Bearer ' +
```
Comment (indungu) [inline authors/tests/test_favorite_article.py]: This hanging indentation does conform to the PEP8 standards but kindly update it to the following for better readability.
```python
response = self.client.put(
    reverse('articles:favorite_article',
        kwargs={'slug': 'kenya-moja'}
    ), content_type='application/json',
    HTTP_AUTHORIZATION='Bearer ' + res.data['token']
)
```

File: authors/tests/test_favorite_article.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        self.assertEqual(response.data["message"],
+                         'Article successfully added to favorites')
+
+    def test_add_unavailable_article_to_favorite(self):
+        """Test if user can add unavailable article to favorites"""
+        res = self.add_article()
+        response = self.client.put(reverse('articles:favorite_article',
+                                           kwargs={'slug': 'kenya-mojaa'}),
+                                   content_type='application/json',
+                                   HTTP_AUTHORIZATION='Bearer ' +
```
Comment (indungu) [inline authors/tests/test_favorite_article.py]: Same as above

File: authors/tests/test_favorite_article.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(response.data["message"], 'Article not found')
+
+    def test_remove_article_from_favorite(self):
+        """Test if user can remove article from favorites"""
+        res = self.add_article()
+        self.client.put(reverse('articles:favorite_article',
+                                kwargs={'slug': 'kenya-moja'}),
+                        content_type='application/json',
+                        HTTP_AUTHORIZATION='Bearer ' +
```
Comment (indungu) [inline authors/tests/test_favorite_article.py]: Same as above

File: authors/tests/test_favorite_article.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                                           res.data['token'])
+
+        response = self.client.put(reverse('articles:favorite_article',
+                                           kwargs={'slug': 'kenya-moja'}),
+                                   content_type='application/json',
+                                   HTTP_AUTHORIZATION='Bearer ' +
```
Comment (indungu) [inline authors/tests/test_favorite_article.py]: Same as above

File: authors/tests/test_favorite_article.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data["message"],
+            'You have successfully removed this article from favorites')
+
+    def test_get_all_favorite_articles(self):
+        """Test if user can get all favorite articles"""
+        res = self.add_article()
+        response = self.client.get(reverse('articles:all_favourite_article')
+                                   , content_type='application/json',
```
Comment (indungu) [inline authors/tests/test_favorite_article.py]: same as above

File: authors/urls.py [hunk 6]
```diff
@@ -29,6 +29,7 @@
     path('api/profiles/', include('authors.apps.profiles.urls',
                                   namespace='profiles')),
     path('api/articles', include('authors.apps.articles.urls',
+
```
Comment (indungu) [inline authors/urls.py]: Please remove unnecessary whitespace characters.

**Review by indungu — CHANGES_REQUESTED** [review]
Nice work. Please make the following adjustments.

**Review by Allan690 — APPROVED** [review]
LGTM

**Review by kwanj-k — APPROVED** [review]

**Review by indungu — APPROVED** [review]
